### DISCUSSION TITLE
Improved waypoints icon set (waypoints)

### DISCUSSION THREAD

**Discussion by elgandoz** [XCSoar/XCSoar#1400] [General]
[As I mentioned in a PR a while ago](https://github.com/XCSoar/XCSoar/pull/972#issuecomment-1250648894), I wanted to improve XCSoar icons. Few of the points I made at that time:

 > - icons do not have a consistent visual weight (some appear prominent, some almost disappear in the background)
 > - some icons are rendered blurry due to the aliasing/scaling, this could be improved by uniforming and simplifying the design
 > - many of them have too many details, making them hard to recognise especially on a phone mounted on a bouncy cockpit.
 > - uniform the general style (some have drop shadow, thin/thick/no borders, tiny/big details), taking inspiration from topographic conventions and trying to maintain the current visuals.
 > - uniform the sizing: every icon should be contained in the same canvas in order to have consistent size
 > - improve contrast, especially for the coloured ones.
 > - Simplify the markup: SVG is an open standard that can be read and edit by application, but many icons contains lots of filters and vendor metatags that could create inconsistent rendering. This could be done by proper export and/or using cli tools like [SVGO](https://github.com/svg/svgo).

Since then a lot of changes have been made by the contributors ( mostly by @MikeBravo15  and @Sundown3867 ), and many of those points have been addressed or at least partially.

I wanted to further improve the icon set, with the goal to make them more visible and aesthetically pleasing. Part of this has been inspired by other flight computer display, like SeeYou Nav and Skytraxx (as rough guideline, all the graphic has been created by me).

Some of the improvements:
- increase contrast by using a bright outline for the icon
- simple, recognisable shape, following well know map conventions
- "pixel perfect" design: all the vector graphics are optimised to be rendered well at the minimum resolution, minimizing anti-aliasing artefacts
- balancing icons in size, depending on "popularity" (more common type like turnpoints, mountains and passes are a little smaller in size). This is meant to improve readability in "crowded" areas.
- Simplified the vector output with minimum bloat of meta (currently in progress, ideally I would like to optimise the shapes using [SVGO](https://github.com/svg/svgo))
- Use of SVG's meta `viewbox` and **sizing** (`width` and `height`) for normalising the output
- In the template, consideration of the sizing and positioning in relation to the label (currently most of the icons overlap, reducing readability)

I would really appreciate some feedback (good or bad), and to collaborate with anyone interested.
After establishing this icon set, I would also like to work on the rest of the icons used throughout the interface.

**Current Preview**
<img src="https://github.com/XCSoar/XCSoar/assets/5859926/b89036b0-eb08-467e-9dc9-aa8d23486d26" alt="XCS new icons Android" width="300" />
[CUP file for demo](https://github.com/XCSoar/XCSoar/files/14921558/XCSoar_Icons_with_labels.cup.zip)

-----

**Current icons details**
| Icons | Pixel preview | Frames and Guides |
|-|-|-|
| ![Screenshot 2024-04-09 at 16 30 27](https://github.com/XCSoar/XCSoar/assets/5859926/06dc76b1-ac83-495d-954e-de3d10cb89d5) | ![Screenshot 2024-04-09 at 16 30 38](https://github.com/XCSoar/XCSoar/assets/5859926/d9c61e6f-93e4-4146-bdfd-6b14b1b99865) | ![Screenshot 2024-04-09 at 16 41 45](https://github.com/XCSoar/XCSoar/assets/5859926/972c0370-d80c-4445-843d-87ae26ce97d5) |

**Icon template details**
<img src="https://github.com/XCSoar/XCSoar/assets/5859926/ebb9efb9-11f5-40b3-80b2-87c2882565fd" alt="icon template details" width="" />
14x14px grey area for approximate icon sizing

-----

**Current working file**
 - https://www.figma.com/file/3LNqQ9XdrxPk4CxGhyHLrU/XCSoar-icons
**NB**: any icon in the repo is a perfectly valid source, using only standard SVG (no vendor metas)

-----

**Attached PR**
- #1394 

Related issues/PRs:
- #972 
- #1130 
- #1155 
- #982 
- #1116 
- #921 
- #1359

**Comment by lordfolken** [comment]
When i added the new icons, i always hoped that someone would find them ugly enough to be motivated to update them. 
So thanks to you and @Sundown3867 . :-)

My most important points:

 - [ ] above all: reproducability. meaning:
        * automation (any process that has to happen has to be automated)
        * methodology
        * guidelines
        * design principles
        Should be documented in the dev manual https://github.com/XCSoar/XCSoar/tree/master/doc
        I see already a lot of that in your opening post here, but: Git is for the ages (this code base starts in 2002). 
        Github is a temporary thing. 
  - [ ] accountability:
          I'd like a team of a couple of people that agree on what things should look like. That exert authority on the process and
          upkeep the quality of potential contributions. (and hopefully rework all the ugly stuff)
          I can add the paths to  https://github.com/XCSoar/XCSoar/blob/master/.github/CODEOWNERS so you get notified of PRs and  issues. and of course create a github team.
       
I'm a guy that lives in the Terminal so I'm probably not that good at graphics. 
Although i do see that design is a very structured approach when  done properly.
So i would be very happy to hand over those parts of responsibility to someone that cares.
